backends/CMakeLists.txt

```diff
@@ -0,0 +1,133 @@
+if(IMGUI_BACKEND_DX11)
+    add_library(imgui_backend_dx11 imgui_impl_dx11.cpp)
+    target_include_directories(imgui_backend_dx11 PUBLIC .)
+    target_link_libraries(imgui_backend_dx11 PUBLIC imgui d3d11.lib d3dcompiler.lib)
+endif()
+
+option(IMGUI_BACKEND_DX12 "Enable the DirectX 12 backend" OFF)
+if(IMGUI_BACKEND_DX12)
+    add_library(imgui_backend_dx12 imgui_impl_dx12.cpp)
+    target_include_directories(imgui_backend_dx12 PUBLIC .)
+    target_link_libraries(imgui_backend_dx12 PUBLIC imgui d3d12.lib d3dcompiler.lib dxgi.lib)
+endif()
+
+option(IMGUI_BACKEND_DX9 "Enable the DirectX 9 backend" OFF)
+if(IMGUI_BACKEND_DX9)
+    add_library(imgui_backend_dx9 imgui_impl_dx9.cpp)
+    target_include_directories(imgui_backend_dx9 PUBLIC .)
+    target_link_libraries(imgui_backend_dx9 PUBLIC imgui d3d9.lib)
+endif()
+
+option(IMGUI_BACKEND_GLFW "Enable the GLFW backends" OFF)
+if(IMGUI_BACKEND_GLFW)
+    find_package(glfw3 REQUIRED)
+    add_library(imgui_backend_glfw imgui_impl_glfw.cpp)
+    target_include_directories(imgui_backend_glfw PUBLIC .)
+    target_link_libraries(imgui_backend_glfw PUBLIC imgui glfw)
+endif()
+
+option(IMGUI_BACKEND_GLUT "Enable the GLUT backend" OFF)
+if(IMGUI_BACKEND_GLUT)
+    find_package(glut REQUIRED)
+    add_library(imgui_backend_glut imgui_impl_glut.cpp)
+    target_include_directories(imgui_backend_glut PUBLIC .)
+    target_link_libraries(imgui_backend_glut PUBLIC imgui glut)
+endif()
+
+option(IMGUI_BACKEND_METAL "Enable the Metal backend" OFF)
+if(IMGUI_BACKEND_METAL)
+    add_library(imgui_backend_metal imgui_impl_metal.mm)
+    target_include_directories(imgui_backend_metal PUBLIC .)
+    target_link_libraries(imgui_backend_metal PUBLIC imgui "-framework Metal -framework MetalKit -framework QuartzCore")
+endif()
+
+option(IMGUI_BACKEND_OPENGL "Enable the OpenGL backends" OFF)
+if(IMGUI_BACKEND_OPENGL)
+    find_package(OpenGL REQUIRED)
+
+    add_library(imgui_backend_opengl2 imgui_impl_opengl2.cpp)
+    target_include_directories(imgui_backend_opengl2 PUBLIC .)
+    target_link_libraries(imgui_backend_opengl2 PUBLIC imgui OpenGL::OpenGL)
+
+    add_library(imgui_backend_opengl3 imgui_impl_opengl3.cpp)
+    target_include_directories(imgui_backend_opengl3 PUBLIC .)
+    target_link_libraries(imgui_backend_opengl3 PUBLIC imgui OpenGL::OpenGL)
+endif()
+
+option(IMGUI_BACKEND_OSX "Enable the OSX backend" OFF)
+if(IMGUI_BACKEND_OSX)
+    add_library(imgui_backend_osx imgui_impl_osx.mm)
+    target_include_directories(imgui_backend_osx PUBLIC .)
+    target_link_libraries(imgui_backend_osx PUBLIC imgui "-framework Cocoa")
+endif()
+
+option(IMGUI_BACKEND_SDL2 "Enable the SDL2 backends" OFF)
+if(IMGUI_BACKEND_SDL2)
+    find_package(SDL2 REQUIRED)
+
+    add_library(imgui_backend_sdl2 imgui_impl_sdl2.cpp)
+    target_include_directories(imgui_backend_sdl2 PUBLIC .)
+    target_link_libraries(imgui_backend_sdl2 PUBLIC imgui SDL2::SDL2main SDL2::SDL2)
```

Comment:
`SDL2::SDL2main` is not necessary here.

Also `SDL2::SDL2` is a shared lib and for static linking, you need to use `SDL2::SDL2-static`, so probably needs to do something like this:

```cmake
if(BUILD_SHARED_LIBS)
  target_link_libraries(imgui_packend_sdl2 PUBLIC
    SDL2::SDL2
  )
else()
  target_link_libraries(imgui_packend_sdl2 PUBLIC
    SDL2::SDL2-static
  )
endif()
```